src/components/quantity-selector/quantity-selector.tsx

```diff
@@ -0,0 +1,69 @@
+import classNames from "classnames";
+import React, { PureComponent } from "react";
+import "../../styles/components/_quantity-selector.scss";
+import Button from "../button";
+import Icon from "../icon";
+
+export default class QuantitySelector extends PureComponent<IQuantitySelectorProps, any> {
+
+    public render() {
+        const count = this.props.count || 0;
+
+        if (count === 0) {
+            return <Button fluid variant="secondary" onClick={this.props.onIncrementClick} size="xsmall">
```

Comment:
You should take "fluid", "variant" and "size" props from outside aswell. The one who uses this component may want to have large button.

Reply:
Futhermore, QuantitySelector can get a prop called as "buttonProps" and you can directly pass it to the Button element

Reply:
this part should be not here. this component responsibility should be only increment and decrement given number. if you need another button for `count === 0` case you handle that in your codebase.

Reply:
Thats another case. The thing that I'm saying is the button cannot be changed if no prop overrides